android-components/components/feature/app-links/src/main/java/mozilla/components/feature/app/links/AppLinksUseCases.kt

```diff
@@ -295,11 +295,46 @@ class AppLinksUseCases(
         @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
         internal var redirectCache: AppLinkRedirectCache? = null
 
+        @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+        internal var userDoNotInterceptCache: MutableMap<Int, Long> = mutableMapOf()
+
         @VisibleForTesting
         internal fun clearRedirectCache() {
             redirectCache = null
         }
 
+        @VisibleForTesting
+        internal fun getCacheKey(url: String, appIntent: Intent?): Int? {
+            return Uri.parse(url)?.let { uri ->
+                when {
+                    appIntent?.component?.packageName != null -> appIntent.component?.packageName
+                    !uri.isHttpOrHttps -> uri.scheme
+                    else -> uri.host // worst case we do not prompt again on this host
+                }.hashCode()
+            }
+        }
+
+        internal fun inUserDoNotIntercept(url: String, appIntent: Intent?): Boolean {
+            val cacheKey = getCacheKey(url, appIntent)
+            val cacheTimeStamp = userDoNotInterceptCache[cacheKey]
+            val currentTimeStamp = SystemClock.elapsedRealtime()
+
+            if (cacheTimeStamp != null &&
+                currentTimeStamp <= cacheTimeStamp + APP_LINKS_DO_NOT_OPEN_CACHE_INTERVAL
+            ) {
+                return true
+            }
+
+            return false
+        }
```

Comment:
drive-by - could this be something like:
```suggestion
return cacheTimeStamp != null && currentTimeStamp <= (cacheTimeStamp + APP_LINKS_DO_NOT_OPEN_CACHE_INTERVAL)
```

Reply:
Good catch.  Updated, thanks.